ide/src/focus_utils.ts

```diff
@@ -0,0 +1,45 @@
+import vscode from 'vscode';
+
+export enum FocusStatusBarState {
```

Comment:
You can just use `FocusStatusBarState = "active" | "inactive" | "error"`. That is, sadly, the more idiomatic Typescript.